MainWindow.cpp

```diff
@@ -165,7 +166,9 @@ MainWindow::MainWindow()
 	m_ignoreSelectionChanges(0),
 	m_ignorePageOrderingChanges(0),
 	m_debug(false),
-	m_closing(false)
+    m_closing(false),
+    m_isMultithreaded(false),
+    m_threadCount(2)
 {
```

Comment:
This is just nitpicking, but the code uses tabs instead of spaces for indentation, would be great if you could adapt your changes to that :-)